src/backends/backend/triton_model_instance.cc

```diff
@@ -580,6 +587,8 @@ TritonModelInstance::Execute(
 
     TRITONSERVER_ErrorDelete(err);
   }
+
+  ongoing_requests_count_--;
```

Comment:
The decrement is done in `Execute()`, if the request is delegated to a separate thread, Does the `count` track the ongoing request properly? In this case, the `inst_exec_fn` returns "immediately", right?

Reply:
It doesn't return immediately. We have subtests that have WAIT time specified as large as 10 sec. That being said I was too much focused on triaging the issue that I came up with this very specific solution. I will share a proper fix soon.